networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -198,7 +200,18 @@ def do_create_router(self, context, router, router_type_id, auto_schedule,
         router_created[routertypeawarescheduler.SHARE_HOST_ATTR] = share_host
         router_created[routerhostingdevice.HOSTING_DEVICE_ATTR] = (
             hosting_device_id)
-        return router_created, r_hd_b_db
+        try:
+            router_db = self._get_router(context, router_created['id'])
+            if gw_info:
+                self._update_router_gw_info(context, router_created['id'],
+                                            gw_info)
+        except Exception:
+            with excutils.save_and_reraise_exception():
+                LOG.exception(_LE("An exception occurred while creating "
+                                  "the router: %s"), router)
+                self.delete_router(context, router_created['id'])
```

Comment:
when the exception happens, right now we still return as usual at line#214.....Is this intended?